### DISCUSSION TITLE
marp-cli error when using markdown-it-plantuml

### DISCUSSION THREAD

**Discussion by alanlivio** [marp-team/marp#433] [Q&A] [answered]
Hello. I followed this [guide to use plantuml](https://github.com/orgs/marp-team/discussions/219) from @yhatt. But I get the error bellow.
Any suggestions? 
Thank you in advance.
Best, 
Alan

engine.js file
```js
const { Marp } = require('@marp-team/marp-core')
module.exports = {
    engine: opts => new Marp(opts).use(require('markdown-it-plantuml'))
}
```

package.json file
```
{
  "dependencies": {
    "@marp-team/marp-cli": "^2.5.0",
    "@marp-team/marp-core": "0.13.1",
    "@marp-team/marpit": "1.4.0",
    "markdown-it-plantuml": "^1.4.1"
  }
}
``` 

plantuml.md file
```markdown
---
marp: true
---
<!-- https://github.com/orgs/marp-team/discussions/219 -->

@startuml
  actor actor
  actor/ "actor/"
  agent agent
  artifact artifact
  boundary boundary
  card card
  circle circle
  cloud cloud
  collections collections
  component component
  control control
  database database
  entity entity
  file file
  folder folder
  frame frame
  hexagon hexagon
  interface interface
  label label
  node node
  package package
  person person
  queue queue
  rectangle rectangle
  stack stack
  storage storage
  usecase usecase
  usecase/ "usecase/"
@enduml
```

command line
```bash
npx marp --engine ./engine.js plantuml.md
```
![image](https://user-images.githubusercontent.com/604808/233408634-f7dbe9d0-9a59-43e6-94d8-d2959825529a.png)

**Comment by yhatt** [comment, marked as answer]
The `engine.js` script you've provided is actually [a configuration file for Marp CLI](https://github.com/marp-team/marp-cli#configuration-file),  not an engine. Therefore, use `-c` instead of `--engine` to pass the setting to CLI.

```
marp -c ./xxxxxx.js your-markdown.md
```

An engine script that is passed by `--engine` must export a JavaScript class or function through `module.exports`.

If it was still not working, try updating every Marp tools to the latest versions. I found from your `package.json` that you seems to be depending on very old versions of Marp tools.

**Reply by alanlivio** [reply]
Thanks @yhatt. I put my used code bellow.

package.json
```json
{
  "dependencies": {
    "@marp-team/marp-cli": "^2.5.0",
    "@marp-team/marp-core": "^3.6.0",
    "@marp-team/marpit": "^2.4.2",
    "markdown-it-plantuml": "latest"
  }
}
```
marp.config.js
```js
const { Marp } = require('@marp-team/marp-core')

module.exports = {
    engine: opts => new Marp(opts).use(require('markdown-it-plantuml'))
}
```

I used the same plantuml.md and then ran `marp plantuml.md`.

![image](https://user-images.githubusercontent.com/604808/233588540-8d0ffccb-cb8e-403b-ac42-d62b354c93bb.png)
